api/router.go

```diff
@@ -65,6 +65,52 @@ func (ar *apiRouter) CreateCredential(w http.ResponseWriter, r *http.Request) er
 	return writeJSONResponse(w, http.StatusCreated, resp, "")
 }
 
+func (ar *apiRouter) GetOperatorInfo(w http.ResponseWriter, r *http.Request) error {
+	// Try to decode the request body.
+	var req OperatorInfoRequest
+	if err := readJSONRequest(w, r, &req); err != nil {
+		return writeJSONError(w, err)
+	}
+
+	ar.logger.Info("Got operator info request",
+		zap.String("address", req.Address),
+		zap.String("msg", req.Msg),
+		zap.String("sig", req.Sig),
+		zap.String("version", req.Version),
+		zap.Int("operator_type", int(req.operatorType)),
+	)
+
+	sig, err := hex.DecodeString(strings.TrimPrefix(req.Sig, "0x"))
+	if err != nil {
+		msg := "invalid signature"
+		return writeJSONError(w, &decodingError{status: http.StatusBadRequest, msg: msg})
+	}
```

Comment:
All of this code is identical to the corresponding logic in `CreateCredential`. If we use the type alias i ask for in `api/messages.go`, we can deduplicate this block by creating a function that calls `readJSONRequest`, and decodes and validates the sig, with return type `error`.

Since the type is aliased, you will have to cast it in the OperatorInfo handler.

Reply:
Great suggestion. That section of the code looks much cleaner now.